Plugins/CoreAPI/server/graphql/schema/mutations.graphql

```diff
@@ -0,0 +1,9 @@
+type Mutation {
+  setCoreAPIConfiguration(hookURL: String!): CoreAPIConfiguration
+  upsertHosts(projectID: ID!, hosts: [HostInput]!): MutationResponse!
```

Comment:
Typically in GraphQL, you would return the responses as the actual type so that the mutation could actually request fields to be returned and update the UI without a follow-on query. You could declare an `interface` type and then implement it per type you want to have a response for. Like:

```graphql
interface MutationResponseInterface {
    status: String
    was_success: Boolean
}

type HostsMutationResponse implements MutationResponseInterface {
    status: String
    was_success: Boolean
    hosts: [Host]
}
```

This gives the mutator the flexibility to request fields if needed and makes the GraphQL schema more explicit. It's more verbose, but that's not a bad thing in GraphQL-land

Reply:
also not a request for action currently, just an education moment